service/worker/deletenamespace/reclaimresources/workflow.go

```diff
@@ -0,0 +1,157 @@
+		WorkflowRunTimeout: 60 * time.Minute,
+	}
+)
+
+func validateParams(params *ReclaimResourcesParams) error {
+	if params.NamespaceID.IsEmpty() {
+		return temporal.NewNonRetryableApplicationError("namespace ID is empty", "", nil)
+	}
+
+	if params.Namespace.IsEmpty() {
+		return temporal.NewNonRetryableApplicationError("namespace is empty", "", nil)
+	}
+
+	return nil
+}
+
+func ReclaimResourcesWorkflow(ctx workflow.Context, params ReclaimResourcesParams) (ReclaimResourcesResult, error) {
+	logger := workflow.GetLogger(ctx)
+	logger.Info("Workflow started.", tag.WorkflowType(WorkflowName))
+
+	if err := validateParams(&params); err != nil {
+		return ReclaimResourcesResult{}, err
+	}
+
+	var a *Activities
+
+	// Step 1. Delete workflow executions.
+	result, err := deleteWorkflowExecutions(ctx, params)
+	if err != nil {
+		return result, err
+	}
+
+	// Step 2. Delete namespace.
+	ctx5 := workflow.WithLocalActivityOptions(ctx, localActivityOptions)
+	err = workflow.ExecuteLocalActivity(ctx5, a.DeleteNamespaceActivity, params.Namespace, params.NamespaceID).Get(ctx, nil)
+	if err != nil {
+		return result, fmt.Errorf("%w: DeleteNamespaceActivity: %v", errors.ErrUnableToExecuteActivity, err)
+	}
+
+	logger.Info("Workflow finished successfully.", tag.WorkflowType(WorkflowName))
+	return result, nil
+}
+
+func deleteWorkflowExecutions(ctx workflow.Context, params ReclaimResourcesParams) (ReclaimResourcesResult, error) {
+	var a *Activities
+	logger := workflow.GetLogger(ctx)
+	deleteAttempt := int32(1)
+	var result ReclaimResourcesResult
+
+	for {
+		ctx1 := workflow.WithChildOptions(ctx, deleteExecutionsWorkflowOptions)
+		var der deleteexecutions.DeleteExecutionsResult
+		err := workflow.ExecuteChildWorkflow(ctx1, deleteexecutions.DeleteExecutionsWorkflow, params.DeleteExecutionsParams).Get(ctx, &der)
+		if err != nil {
+			logger.Error("Unable to execute child workflow.", tag.WorkflowType(deleteexecutions.WorkflowName), tag.Error(err))
+			return result, fmt.Errorf("%w: %s: %v", errors.ErrUnableToExecuteChildWorkflow, deleteexecutions.WorkflowName, err)
+		}
+		result.SuccessCount += der.SuccessCount
+		result.ErrorCount += der.ErrorCount
+
+		checkExecutionsActivityOptions := workflow.ActivityOptions{
+			// 445 seconds of total retry interval.
+			RetryPolicy: &temporal.RetryPolicy{
+				InitialInterval:    1 * time.Second,
+				MaximumInterval:    200 * time.Second,
+				BackoffCoefficient: 1.8,
+				MaximumAttempts:    10,
+			},
+			StartToCloseTimeout:    30 * time.Second,
+			ScheduleToCloseTimeout: 600 * time.Second,
+		}
+		ctx2 := workflow.WithActivityOptions(ctx, checkExecutionsActivityOptions)
+		err = workflow.ExecuteActivity(ctx2, a.EnsureNoExecutionsActivity, params.NamespaceID, params.Namespace).Get(ctx, nil)
+		if err == nil {
+			break
+		}
+
+		if !stderrors.Is(err, errors.ErrExecutionsStillExist) {
+			return result, fmt.Errorf("%w: CheckExecutionsExistActivity: %v", errors.ErrUnableToExecuteActivity, err)
+		}
```

Comment:
Please add a unit test to cover this case. Errors returned from activity will be wrapped as ActivityError, I want to make sure the wrapping would still work with this code.